tests/test_runner/network_tests.py

```diff
@@ -0,0 +1,128 @@
+        self.subproc = subprocess.Popen(["netcat",
+                                         "-l",
+                                         "localhost",
+                                         "7357"
+                                        ],
+                                        universal_newlines=True,
+                                        stdout=subprocess.PIPE,
+                                        stderr=subprocess.PIPE)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, etype, value, traceback):
+        self.subproc.kill()
+
+    def get_output(self):
+        # kill the process
+        self.subproc.kill()
+
+        # read stdout and return it
+        output = self.subproc.communicate()
+        return output[0]
+
+
+def execute_test():
+
+    # We use a preconfigured tarball bundle for this test, the container has the following arg:
+    #
+    #   echo DOBBY_TEST | netcat 100.64.11.1 7357
+    #
+    # This will send 'DOBBY_TEST' to the dobby0 bridge device at port 7357. We're expecting
+    # this to be received on the listening port 7357 on the localhost because of the added
+    # networking configuration in the bundle config:
+    #
+    #   "portForwarding": {
+    #       "containerToHost": [
+    #           {
+    #               "port": 7357,
+    #               "protocol": "udp"
+    #           },
+    #           {
+    #               "port": 7357,
+    #               "protocol": "tcp"
+    #           }
+    #       ]
+    #   }
+    #
+
+    output_table = []
+
+    with test_utils.dobby_daemon(), netcat_listener() as nc, test_utils.untar_bundle(container_name) as bundle_path:
+        # Test 0
+        test = tests[0]
+        command = ["DobbyTool",
+                   "start",
+                   container_name,
+                   bundle_path]
+
+        status = test_utils.run_command_line(command)
+
+        message = ""
+        result = True
+
+        # give container time to start and send message before checking netcat listener
+        sleep(2)
+
+        nc_message = nc.get_output().rstrip("\n")
+
+        # check if netcat listener received message
+        if test.expected_output.lower() not in nc_message.lower():
+            message = "Received '%s' from container, expected '%s'" % (nc_message.lower(), test.expected_output.lower())
+            result = False
+        else:
+            message = "Successfully received message '%s' from container" % nc_message
+
+        output = test_utils.create_simple_test_output(test, result, message, status.stderr)
+        output_table.append(output)
+        test_utils.print_single_result(output)
+
+    return test_utils.count_print_results(output_table)
+
+
+if __name__ == "__main__":
+    test_utils.parse_arguments(__file__)
+
+    # netcat is not available on RDK builds, skip test
+    if test_utils.selected_platform == test_utils.Platforms.xi_6:
```

Comment:
This should be done inside of "execute_test" with return value:
```
if test_utils.selected_platform == test_utils.Platforms.xi_6:
    return test_utils.print_unsupported_platform(basename(__file__), test_utils.selected_platform)
```
The reason for it is that if someone wanted to include this test into some testgroup only "execute_test" will be called, and tester would not know that it is not supported in selected platform.
You can compare it to "thunder_plugin.py".